pkg/util/status/build_status_test.go

```diff
@@ -61,14 +61,14 @@ func TestUpdateStageDuration(t *testing.T) {
 	buildInfo := new(api.BuildInfo)
 
 	buildInfo.Stages = api.RecordStageAndStepInfo(buildInfo.Stages, api.StagePullImages, api.StepPullPreviousImage, time.Now(), time.Now())
-	duration := buildInfo.Stages[0].Duration
+	duration := buildInfo.Stages[0].DurationMilliseconds
 
 	addDuration, _ := time.ParseDuration("5m")
 
 	buildInfo.Stages = api.RecordStageAndStepInfo(buildInfo.Stages, api.StagePullImages, api.StepPullBuilderImage, time.Now(), time.Now().Add(addDuration))
 
-	if !(buildInfo.Stages[0].Duration > duration) {
-		t.Errorf("Stage Duration was not updated, expected %#v, got %#v", buildInfo.Stages[0].StartTime.Add(duration), buildInfo.Stages[0].Duration)
+	if !(buildInfo.Stages[0].DurationMilliseconds > duration) {
+		t.Errorf("Stage Duration was not updated, expected %#v, got %#v", buildInfo.Stages[0].StartTime.Add(time.Duration(duration)), buildInfo.Stages[0].DurationMilliseconds)
```

Comment:
your duration variable here is a millisecond value, converting it to a time.Duration doesn't make sense unless you first scale it back to nanoseconds.